### PR TITLE
Feature/change interview time

### PR DESCRIPTION
**Pull request by FriedSock**
A few things missing from this being a workable story, we don't have the show page for defence requests yet, so solicitors will not be able to view the interview start time. It has also occurred to me that solicitors don't even get an email when they are assigned the case for the first time. There are also some problems with the form at the time of writing before I embark on the mother of all rebases.

### CODE REVIEW

File: app/policies/defence_request_policy.rb [hunk 1]
```diff
@@ -58,6 +58,10 @@ def dscc_number_edit?
     user.cco? && record.opened?
   end
 
+  def interview_start_time_edit?
+    user.cso? && !record.new_record? && record.created?
+  end
```
Comment (dp6ai) [inline app/policies/defence_request_policy.rb]: do we want to be able to add this to a new record?

File: app/mailers/mailer.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Mailer < ApplicationMailer
+  default from: "<DSDS Notifications> noreply@#{Settings.action_mailer.smtp_settings.domain}"
+
+  def notify_interview_start_change(defence_request, solicitor)
+    @solicitor = "solicitor"
+    @defence_request = defence_request
+    mail(to: solicitor.email,
+         subject: 'Interview start time change',
+         content_type: 'text/html',
```
Comment (dp6ai) [inline app/mailers/mailer.rb]: PLAIN TEXT!!

File: app/views/mailer/notify_interview_start_change.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<h1> An interview start time has changed! <h1>
+
+<p>A change has been made to the interview start time for a defence request you are assigned to, please view the new time at (a show url is totally going here when it is actually implemented)</p>
```
Comment (dp6ai) [inline app/views/mailer/notify_interview_start_change.html.erb]: should we include the start time in here?

File: spec/mailers/mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+RSpec.describe Mailer, :type => :mailer do
+  subject { Mailer }
+  let(:defence_request) { FactoryGirl.create(:defence_request) }
+  let(:solicitor) { FactoryGirl.create(:solicitor_user) }
+
+  describe 'notify_interview_start_change' do
+    it 'contains a link to the request' do
+      response = subject.notify_interview_start_change(defence_request, solicitor).deliver_now
+      expect(response.body).to have_content 'a show url is totally going here when it is actually implemented'
```
Comment (dp6ai) [inline spec/mailers/mailer_spec.rb]: Add a TODO

File: config/environments/production.rb [hunk 5]
```diff
@@ -13,6 +13,8 @@
   # Full error reports are disabled and caching is turned on.
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
+  config.action_mailer.delivery_method = :smtp
```
Comment (dp6ai) [inline config/environments/production.rb]: can we disable delivery in sendgrid?

File: app/views/defence_requests/_form.html.erb [hunk 6]
```diff
@@ -25,11 +25,15 @@
...
     <%= f.text_field :solicitor_name, label: t('solicitor_name'), autofocus: true, class: 'form-control' %>
     <%= f.text_field :solicitor_firm, label: t('solicitor_firm'), class: 'form-control' %>
+    <%= f.text_field :solicitor_email, type: :hidden %>
     <%= f.label :scheme, t('scheme'), class: 'form-label-bold' %>
     <%= f.select :scheme, solicitor_schemes, class: 'form-control' %>
     <%= f.text_field :phone_number, label: t('phone_number'), class: 'form-control' %>
     <%= f.text_field :custody_number, label: t('custody_number'), class: 'form-control' %>
     <%= f.text_field :allegations, label: t('allegations'), class: 'form-control' %>
+    <% if policy(@defence_request).interview_start_time_edit? %>
+      <%= f.time_select :interview_start_time, { label: t('interview_start_time') }, { class: 'time-select' } %>
```
Comment (dp6ai) [inline app/views/defence_requests/_form.html.erb]: Needs to have date, as discussed

Comment (FriedSock) [inline app/views/defence_requests/_form.html.erb]: Going to become a text field in the next story.

File: config/environments/production.rb (hunk 5 above)
Comment (FriedSock) [inline config/environments/production.rb]: All emails are example.com, going to setup a gmail and interceptor for testing.

**Pull request closed without merging**